File: backend/app/schemas/clip.py
from pydantic import BaseModel
from typing import Optional, List

class CaptionSchema(BaseModel):
    id: str
    start: float
    end: float
    text: str

    class Config:
        from_attributes = True

class ClipResponse(BaseModel):
    id: str
    job_id: str
    title: Optional[str]
    start_time: float
    end_time: float
    duration: float
    viral_score: float
    file_url: Optional[str]
    thumbnail_url: Optional[str]
    platform_format: str
    captions: List[CaptionSchema] = []

    class Config:
        from_attributes = True

File: frontend/src/types/clip.ts
export interface Clip {
  id: string
  job_id: string
  title: string
  start_time: number
  end_time: number
  duration: number
  interest_score?: number | null
  is_recommended?: boolean
  viral_score?: number
  file_url: string
  thumbnail_url: string
  captions: Caption[]
  broll_segments: BRollSegment[]
  platform_format: "9:16" | "16:9" | "1:1"
}

export interface Caption {
  id: string
  start: number
  end: number
  text: string
}

export interface BRollSegment {
  id: string
  start: number
  end: number
  asset_url: string
  asset_type: "video" | "image"
}
